graphql_api/types/commit/commit.py

```diff
@@ -39,9 +41,22 @@
 
 @commit_bindable.field("coverageFile")
 @sync_to_async
-def resolve_file(commit, info, path, flags=None):
-    commit_report = commit.full_report.filter(flags=flags)
-    file_report = commit_report.get(path)
+def resolve_file(commit, info, path, flags=None, components=None):
+    _else, paths = None, []
+    if components:
+        paths = []
+        all_components = components_service.commit_components(
+            commit, info.context["request"].current_owner
+        )
+        filtered_components = components_service.filter_components_by_name(
+            all_components, components
+        )
+        for fc in filtered_components:
+            paths.extend(fc.paths)
+        _else = FilteredReportFile(ReportFile(path), [])
+
+    commit_report = commit.full_report.filter(flags=flags, paths=paths)
+    file_report = commit_report.get(path, _else=_else)
 
     return {
```

Comment:
It might make sense to also return components here as part of the dict similar to how we are returning flags.